go/graphd/query.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The query portion of the graphd package contains request and response structures used in IO
+// operations.
+
+package graphd
+
+import (
+	"strings"
+)
+
+// A Request to be sent to a graphd database.
+// Currently this is just a wrapper around a string.
+type Request struct {
+	body string
+}
+
+// String implements stringer interface for a Request.
+func (r *Request) String() string {
+	return strings.TrimSpace(r.body)
+}
+
+// A Response received from a graphd database.
+// Currently this is just a wrapper around a string.
+type Response struct {
```

Comment:
if you just want the type safety, you can type alias, too

eg:
`type Response = string`

Reply:
(ditto for Request)

Reply:
Now that I think about it, it might be interesting to make Request/Response both instances of Reader/Writer? eg https://golang.org/pkg/io/#Writer, https://golang.org/pkg/io/#Reader or some sub-specialization

Reply:
This was more like scaffolding for when the Request/Response turns into something other than a simple string.

I'll have a look at Reader/Writer.